provider/machinepool/machine_pool_resource.go

```diff
@@ -997,6 +1012,11 @@ func (r *MachinePoolResource) populateState(object *cmv1.MachinePool, state *Mac
 	} else {
 		state.SubnetID = types.StringValue("")
 	}
+	subnetListValue, err := common.StringArrayToList(subnets)
+	if err != nil {
+		return err
+	}
+	state.AWSSubnetIDs = subnetListValue
```

Comment:
I think that this should be done only in case `len(subnets) != 1`, to avoid having two confusing similar attributes